### PR TITLE
Fix for issue 1760: Updated Student Interface Spreadsheet marks table to React

### PR DESCRIPTION
**Pull request by peterzhao333**
Fix for issue #1760 : 
Updated student interface's spreadsheet marks table to React

### CODE REVIEW

**Comment by peterzhao333** [comment]
Two of the grade_entry_forms_controller_tests seem to be failing because they are using a regular expression to search for the expected result string from the response of a student_interface GET request.

However, because we are using React tables now, I believe they are not compiling in the test and the table data is not updated, and a matching string is not found.

Any ideas what we should do with these @david-yz-liu?

File: app/views/grade_entry_forms/student_interface.html.erb [hunk 1]
```diff
@@ -12,42 +12,136 @@
...
-                  <% end %>
-                </td>
-              <% end %>
-            </tr>
-          </tbody>
-       </table>
-      </div>
+      <div id='marks_table'></div>
+
+      <script type='text/jsx'>
```
Comment (arkon) [inline app/views/grade_entry_forms/student_interface.html.erb]: Should this all be moved to a separate file...?

Comment (peterzhao333) [inline app/views/grade_entry_forms/student_interface.html.erb]: Thanks for the suggestion @arkon , I moved it to its own separate file.

**Comment by david-yz-liu** [comment]
@peterzhao333 I'm okay removing those asserts (not the full tests, just the RegEx asserts).

Everything's looking good, except the spreadsheet columns should be sortable.

**Comment by david-yz-liu** [comment]
Thanks, @peterzhao333!

**Pull request merged**